pkg/scheduler/framework/session.go

```diff
@@ -279,6 +297,46 @@ func (ssn *Session) Allocate(task *api.TaskInfo, hostname string) error {
 	}
 
 	if ssn.JobReady(job) {
+
+		// patch all the pod with backfill annotation. if any fails, abort the backfill
+		var wg sync.WaitGroup
+		errChannel := make(chan error, 1)
+		finished := make(chan bool, 1)
+
+		for _, pendingTask := range job.TaskStatusIndex[api.Allocated] {
+			if pendingTask.Condition.IsBackfill {
+
+				wg.Add(1)
+				go func(pendingTask *api.TaskInfo) {
+					defer wg.Done()
+
+					annotation := map[string]string{v1alpha1.BackfillAnnotationKey: "true"}
+					err := ssn.PatchAnnotation(pendingTask, annotation)
```

Comment:
The annotation should be also cached, there maybe out of sync between scheduler cache & apiserver.

Reply:
WIP